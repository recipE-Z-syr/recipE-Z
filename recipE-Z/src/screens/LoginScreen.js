import 'react-native-gesture-handler';
import * as React from 'react';
import { Platform, SafeAreaView, Button, StyleSheet, Text, View, AppRegistry, Image, TextInput, Alert, ScrollView, Keyboard, TouchableOpacity, Component } from 'react-native';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import defaultStyles from './stylesheet';


class LoginScreen extends React.Component {
  constructor(props) {
    super(props);

    this.state = { email: '', password: ''}

    this.handleEmail = this.handleEmail.bind(this);
    this.handlePassword = this.handlePassword.bind(this);
  }

  handleEmail(text) {
    this.setState({ email: text });
  }

  handlePassword(text) {
    this.setState({ passowrd: text });
  }

  render(){
    const { navigation } = this.props;
  return (
    <View style={[defaultStyles.container, {backgroundColor:'#ffcece'}]}>
    <View style={defaultStyles.whiteContainer}>
      <Image
      style={{width: 90, height: 87}}
      source={require('../img/logo.png')}
      />
      <View style={{alignItems: 'center', justifyContent: 'space-around'}}>
        <Text style={defaultStyles.h1}>
        Welcome Back
        </Text>
        <TouchableOpacity
          onPress={() => navigation.navigate('Sign-Up') } >
          <Text style={[defaultStyles.textLink]}>Don't have an account? Click here to sign up</Text>
        </TouchableOpacity>
      </View>
      <View>
        <TextInput
        placeholder = "Email"
        style={defaultStyles.textInput}
        onBlur = {Keyboard.dismiss}
        value = {this.state.email}
        onChangeText = {this.handleEmail}
        />
        <TextInput
        placeholder = "Password"
        style={defaultStyles.textInput}
        onBlur = {Keyboard.dismiss}
        value = {this.state.password}
        onChangeText = {this.handlePassword}
        />
      </View>
      <TouchableOpacity
        onPress={() => navigation.navigate('Sign-Up')}>
        <Text style={defaultStyles.textLink}>Forgot Password?</Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={[defaultStyles.redButton, {}]}
        onPress={() => navigation.navigate('Search')}
        underlayColor='#ed4848'>
        <Text style={defaultStyles.redButtonText}>LOGIN</Text>
      </TouchableOpacity>
      </View>
    </View>


  );
}
}



export default LoginScreen;

// const styles = StyleSheet.create({
// })
