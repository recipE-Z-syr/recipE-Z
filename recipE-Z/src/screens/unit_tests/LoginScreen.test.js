//LoginScreen.test.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, ShallowWrapper } from "enzyme";

import LoginScreen from '../LoginScreen';

describe('<LoginScreen />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<LoginScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should change state if email is entered', () => {
  	const instanceOf = renderer.create(<LoginScreen />).getInstance();
  	instanceOf.handleEmail('abc@gmail.com');
  	expect(instanceOf.state.email).toEqual('abc@gmail.com');
  });
  it('should change state if password is entered', () => {
  	const instanceOf = renderer.create(<LoginScreen />).getInstance();
  	instanceOf.handleEmail('password123');
  	expect(instanceOf.state.email).toEqual('password123');
  });
});

const createTestProps = () => ({
  navigation: {
    navigate: jest.fn()
  }
});

test('No account navigates to Sign Up', () => {
  // const spy = jest.spyOn(navigation, 'navigate')
  let wrapper: ShallowWrapper;
  let props: any
  props = createTestProps();
  wrapper = shallow(<LoginScreen {...props} />);  

  const startButton = wrapper.find('TouchableOpacity').at(0)
  startButton.props().onPress()

  expect(props.navigation.navigate).toBeCalledWith('Sign-Up')
})

test('Login navigates to Search Screen', () => {
  // const spy = jest.spyOn(navigation, 'navigate')
  let wrapper: ShallowWrapper;
  let props: any
  props = createTestProps();
  wrapper = shallow(<LoginScreen {...props} />);  

  const startButton = wrapper.find('TouchableOpacity').at(2)
  startButton.props().onPress()

  expect(props.navigation.navigate).toBeCalledWith('Search')
})