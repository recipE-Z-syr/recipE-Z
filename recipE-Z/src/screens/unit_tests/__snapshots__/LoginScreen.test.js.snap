// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "fontFamily": "nunito-regular",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#ffcece",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "height": 561,
        "justifyContent": "space-around",
        "paddingBottom": 70,
        "paddingTop": 70,
        "position": "absolute",
        "width": 334,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 87,
          "width": 90,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "space-around",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ed4848",
            "fontFamily": "nunito-extrabold",
            "fontSize": 20,
          }
        }
      >
        Welcome Back
      </Text>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ed4848",
                "fontFamily": "nunito-regular",
                "fontSize": 11,
                "textAlign": "center",
              },
            ]
          }
        >
          Don't have an account? Click here to sign up
        </Text>
      </View>
    </View>
    <View>
      <TextInput
        allowFontScaling={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Email"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "lightgrey",
            "borderWidth": 1,
            "fontFamily": "nunito-regular",
            "height": 50,
            "marginBottom": 5,
            "marginTop": 5,
            "opacity": 0.7,
            "paddingLeft": 15,
            "width": 280,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Password"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "lightgrey",
            "borderWidth": 1,
            "fontFamily": "nunito-regular",
            "height": 50,
            "marginBottom": 5,
            "marginTop": 5,
            "opacity": 0.7,
            "paddingLeft": 15,
            "width": 280,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ed4848",
            "fontFamily": "nunito-regular",
            "fontSize": 11,
            "textAlign": "center",
          }
        }
      >
        Forgot Password?
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ed4848",
          "borderColor": "#ed4848",
          "borderRadius": 5,
          "borderWidth": 10,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingTop": 10,
          "width": 200,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "nunito-bold",
            "paddingLeft": 10,
            "paddingRight": 10,
            "textAlign": "center",
          }
        }
      >
        LOGIN
      </Text>
    </View>
  </View>
</View>
`;
